server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingUtils.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ *  Utils for Weighted Routing
+ *
+ * @opensearch.internal
+ */
+public class WeightedRoutingUtils {
+
+    /**
+     * This function checks if the node is weighed away ie weighted routing weight is set to 0,
+     *
+     * @param nodeId the node
+     * @return true if the node has attribute value with shard routing weight set to zero, else false
+     */
+    public static boolean isWeighedAway(String nodeId, ClusterState clusterState) {
+        DiscoveryNode node = clusterState.nodes().get(nodeId);
```

Comment:
What is the node has dropped during this check? Line 45 will throw `NullPointerException`

Reply:
makes sense to put a null check. I think in this case we need to consider node as weigh away since we can't explicitly check for this. What are your thoughts?